File: ClinicSystem/src/com/example/clinicsystem/helpers/DatabaseConnection.java
package com.example.clinicsystem.helpers;

import java.sql.*;

public class DatabaseConnection {
    private String host = "jdbc:mysql://localhost:3306/clinic";
    private String uName = "root";
    private String uPass = "Awsomemosta12232";
    private Connection connection;
    private static DatabaseConnection INSTANCE = null;

    private DatabaseConnection() {
        try {
            connection = DriverManager.getConnection(host, uName, uPass);
        } catch (SQLException e) {
            System.out.println("Connection Error: ");
            e.printStackTrace();
        }
    }

    public static DatabaseConnection getINSTANCE() {
        if(INSTANCE == null) {
            INSTANCE = new DatabaseConnection();
        }

        return INSTANCE;
    }

    public ResultSet select(String table, String condition) {
        ResultSet resultSet;

        try {
            Statement statement = connection.createStatement();
            String query = "SELECT * FROM " + table + condition;
            resultSet = statement.executeQuery(query);

            return resultSet;
        } catch (SQLException e) {
            System.out.println("Select Query Error: ");
            e.printStackTrace();
            return null;
        }
    }

    public int insertInto(String table, String columns, String values) {
        try {
            Statement statement = connection.createStatement();
            String query = "INSERT INTO " + table + " " + columns + " VALUES " + values;
            int id = statement.executeUpdate(query, Statement.RETURN_GENERATED_KEYS);

            ResultSet resultSet = statement.getGeneratedKeys();

            if (resultSet.next()) {
                id = resultSet.getInt(1);
            }

            return id;
        } catch (SQLException e) {
            System.out.println("Insert Query Error: ");
            e.printStackTrace();
        }

        return -1;
    }

    public void update(String table, String newValues, int id, String idColumn) {
        try {
            Statement statement = connection.createStatement();
            String query = "UPDATE " + table + " SET " + newValues + " WHERE " + idColumn + " = " + id;

            statement.executeUpdate(query);
        } catch (SQLException e) {
            System.out.println("Update Query Error: ");
            e.printStackTrace();
        }
    }
}

File: clinic/DatabaseConnection.php
<?php

class DatabaseConnection {
	private $host="localhost";
	private $user="root";
	private $password="";
	private $dbname="clinic";
	private static $INSTANCE;
	private $connection;
	
	private function __construct() {
		$conn = mysqli_connect($this->host, $this->user, $this->password, $this->dbname);
		
		if($conn) {
			$this->connection=$conn;
		} else {
			die("Connection failed: " . mysqli_connect_error());
		}
	}
	
	public static function getINSTANCE() {
		if (self::$INSTANCE == null) {
			self::$INSTANCE = new DatabaseConnection();
		}
		
		return self::$INSTANCE;
	}
	
	public function select($table, $condition) {
		$query = "SELECT * FROM " . $table . $condition;
		$resultSet = mysqli_query($this->connection, $query);
		
		if($resultSet != null) {
			return $resultSet;
		} else {
			//echo "No Data Available.";
			return null;
		}
		
    }

    public function insertInto($table, $columns, $values) {
		$query = "INSERT INTO " . $table . " " . $columns . " VALUES " . $values;
		
		if ($this->connection->query($query) == TRUE) {
			return $this->connection->insert_id;
		} else {
			echo "Insert unsuccessful: " . $this->connection->error . "<br>";
		}
    }

    public function update($table, $newValues, $id, $idColumn) {
		$query = "UPDATE " . $table . " SET " . $newValues . " WHERE " .
			$idColumn . " = " . $id;
			
		if ($this->connection->query($query) == TRUE) {
				
		} else {
			echo "Insert unsuccessful: " . $this->connection->error . "<br>";
		}   
    }
}

?>